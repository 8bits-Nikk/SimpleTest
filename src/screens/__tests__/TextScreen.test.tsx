import React from 'react';
import TextScreen from '../TextScreen';
import 'react-native';
import {renderWithNavigation} from '../../helper/renderWithNavigation';

describe('TextScreen tests', () => {
  it('should render correctly', () => {
    const tree = renderWithNavigation(<TextScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have text', () => {
    const tree = renderWithNavigation(<TextScreen />);
    const text = tree.queryByText('Hello Tests');
    expect(text).not.toBeNull();
  });

  it('should have text find by TestId', () => {
    const tree = renderWithNavigation(<TextScreen />);
    const text = tree.queryByTestId('text');
    expect(text?.children[0]).toBe('Hello Tests');
  });
});
