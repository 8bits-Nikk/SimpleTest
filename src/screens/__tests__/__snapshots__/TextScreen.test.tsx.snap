// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextScreen tests should render correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#123456",
          "fontSize": 32,
        }
      }
      testID="text"
    >
      Hello Tests
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Previous_btn"
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Previous
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Next_btn"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
